### PR TITLE
bpo-40335: Correctly handle multi-line strings in tokenize error scenarios

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue40335

### CODE REVIEW

**Comment by pablogsal** [comment]
CC: @asottile Can you check if this PR runs correctly with your test suite?

**Comment by asottile** [comment]
yep pyflakes is now passing (with a minor patch, but this patch is expected):

```diff
$ git diff
diff --git a/pyflakes/test/test_api.py b/pyflakes/test/test_api.py
index 128aa69..b579ac8 100644
--- a/pyflakes/test/test_api.py
+++ b/pyflakes/test/test_api.py
@@ -449,6 +449,8 @@ def baz():
         with self.makeTempFile(source) as sourcePath:
             if PYPY:
                 message = 'end of file (EOF) while scanning triple-quoted string literal'
+            elif sys.version_info >= (3, 9):
+                message = 'invalid string prefix'
             else:
                 message = 'invalid syntax'
 
```

**Comment by pablogsal** [comment]
> but this patch is expected

Yeah, that was on purpose 😉

**Comment by lysnikolaou** [comment]
Thanks, @pablogsal, for the fix!

File: Lib/test/test_exceptions.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ def check(src, lineno, offset, encoding='utf-8'):
                 if not isinstance(src, str):
                     src = src.decode(encoding, 'replace')
                 line = src.split('\n')[lineno-1]
-                self.assertEqual(cm.exception.text.rstrip('\n'), line)
+                self.assertIn(line, cm.exception.text.rstrip('\n'))
```
Comment (gvanrossum) [inline Lib/test/test_exceptions.py]: You don't really need the `rstrip()` any more.
```suggestion
                self.assertIn(line, cm.exception.text)
```

**Review by gvanrossum — APPROVED** [review]
Weak approve: I don't feel super confident with the ins and outs of this code, but if the tests pass, I feel it's fine.

**Pull request merged**